src/apps/logging/logging.cpp

```diff
@@ -257,6 +259,81 @@ namespace loggingapp
       };
       // SNIPPET_END: log_record_text
 
+      auto& historical_state = context.get_historical_state();
+      auto get_historical = [this, &historical_state](ccf::RequestArgs& args) {
+        const auto params =
+          nlohmann::json::parse(args.rpc_ctx->get_request_body());
+        const auto in = params.get<LoggingGetHistorical::In>();
+
+        // Check that the requested transaction ID is committed
+        {
+          const auto tx_view = consensus->get_view(in.seqno);
+          const auto committed_seqno = consensus->get_committed_seqno();
+          const auto committed_view = consensus->get_view(committed_seqno);
+
+          const auto tx_status = ccf::get_tx_status(
+            in.view, in.seqno, tx_view, committed_view, committed_seqno);
+          if (tx_status != ccf::TxStatus::Committed)
+          {
+            args.rpc_ctx->set_response_status(HTTP_STATUS_BAD_REQUEST);
```

Comment:
Debatable, but I would expect "404: No such committed transaction". 400 is generally some sort of request validation failure (invalid syntax, request too large etc).

Reply:
We should probably do a pass on our error codes - we're using 400 as a fairly general "your request didn't meet our expectations". We can change some of these to 404s if we've _parsed_ the request, but don't have an answer. I was thinking a 404 that varies with the request body seems odd, but I guess the response body makes it clear what kind of 404 this is.